### PR TITLE
[IOTDB-2747] Start data node

### PR DESCRIPTION
**Pull request by MrQuansy**
- data node entrance
- script of start and stop data node
- configuration for data node
- thrift server of internal service
- connect with config node to join into the cluster

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/CommonUtils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if (split.length != 2) {
+      logger.warn("Bad node url: {}", nodeUrl);
+      return null;
+    }
+    String ip = split[0];
+    try {
+      int port = Integer.parseInt(split[1]);
+      result.setIp(ip).setPort(port);
+    } catch (NumberFormatException e) {
+      logger.warn("Bad node url: {}", nodeUrl);
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/CommonUtils.java:45]: return null?

File: server/pom.xml [hunk 2]
```diff
@@ -82,6 +82,16 @@
             <artifactId>influxdb-thrift</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.iotdb</groupId>
+            <artifactId>iotdb-thrift-datanode</artifactId>
+            <version>0.14.0-SNAPSHOT</version>
```
Comment (OneSizeFitsQuorum) [inline server/pom.xml:88]: ${project.version}

File: server/pom.xml [hunk 3]
```diff
@@ -82,6 +82,16 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iotdb</groupId>
+            <artifactId>iotdb-thrift-confignode</artifactId>
+            <version>0.14.0-SNAPSHOT</version>
```
Comment (OneSizeFitsQuorum) [inline server/pom.xml:93]: ${project.version}

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 4]
```diff
@@ -803,8 +808,28 @@
...
+  private List<String> configNodeUrls;
+
+  /** Internal ip for data node */
+  private String internalIp;
+
+  /** Internal port of data node */
+  private int internalPort = 9003;
+
+  /** The max time of data node waiting to join into the cluster */
+  private long joinClusterTimeOutMs = TimeUnit.SECONDS.toMillis(5);
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:823]: Maybe large? such as 1 minute?

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 5]
```diff
@@ -131,7 +135,14 @@ private void loadProps() {
       properties.load(inputStream);
 
       conf.setRpcAddress(properties.getProperty("rpc_address", conf.getRpcAddress()));
-      replaceHostnameWithIP();
+
+      loadClusterProps(properties);
+
+      try {
+        replaceHostnameWithIP();
```
Comment (OneSizeFitsQuorum) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:142]: Maybe we do not need this anymore if we can use FQDN to identify our nodes

**Review by OneSizeFitsQuorum — COMMENTED** [review]
PTAL. Many class names and class dependencies need to be discussed in detail with MPP query module

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
Better to renane  all `DataNodeInternalServiceXXX` to `DataNodeManagementService`, internal servcei will be used to refer to rpc service between data nodes.

File: server/src/assembly/resources/sbin/start-node.bat [hunk 6]
```diff
@@ -0,0 +1,136 @@
+@REM
```
Comment (chengjianyun) [inline server/src/assembly/resources/sbin/start-node.bat:1]: rename the file to start-data-node.bat?

Comment (MrQuansy) [reply server/src/assembly/resources/sbin/start-node.bat:1]: ok

**Review by HTHou — APPROVED** [review]

**Review by JackieTien97 — APPROVED** [review]

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
Don't change the previous TsServiceImpl

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**